src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.PipeReader.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections;
+using System.IO.Pipelines;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public partial class HttpConnection
+    {
+        private class HttpConnectionPipeReader : PipeReader
+        {
+            private readonly HttpConnection _connection;
+
+            public HttpConnectionPipeReader(HttpConnection connection)
+            {
+                _connection = connection;
+            }
+
+            public override void AdvanceTo(SequencePosition consumed)
+            {
+                _connection._transportChannel.Input.AdvanceTo(consumed);
```

Comment:
This will NRE if the connection hasn't been started. We should put in a check that provides a better error message.

Reply:
I don’t want to put checks in all of the methods here since some of performance critical. I’ll see what makes sense and update.

Reply:
`catch(NullReferenceException)` :trollface: